File: backend/app/schemas/publications.py
"""Publication-related schemas."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class PublicationSummaryItem(BaseModel):
    """Single publication audit row for list pages."""

    id: int
    article_id: int
    article_title: str | None = None
    article_slug: str | None = None
    article_publish_status: int | None = None
    platform: str
    publish_mode: str
    status: str
    trigger_mode: str
    attempt_no: int
    retry_of_publication_id: int | None = None
    external_id: str | None = None
    external_url: str | None = None
    message: str | None = None
    error_message: str | None = None
    published_at: datetime | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class PublicationDetailItem(PublicationSummaryItem):
    """Detailed publication payload."""

    request_payload_json: dict[str, Any] | list[Any] | None = None
    response_payload_json: dict[str, Any] | list[Any] | None = None
    retry_attempts_total: int = 0


class PublicationListPayload(BaseModel):
    """Paginated publication list payload."""

    items: list[PublicationSummaryItem] = Field(default_factory=list)
    total: int
    limit: int
    offset: int
    warning: str | None = None


class PublicationDetailPayload(BaseModel):
    """Single publication response."""

    publication: PublicationDetailItem | None = None
    warning: str | None = None

File: frontend_v2/src/types/publications.ts
import type { PaginatedPayload } from './api'

export interface PublicationSummaryItem {
  id: number
  article_id: number
  article_title: string | null
  article_slug: string | null
  article_publish_status: number | null
  platform: string
  publish_mode: string
  status: string
  trigger_mode: string
  attempt_no: number
  retry_of_publication_id: number | null
  retryable: boolean
  external_id: string | null
  external_url: string | null
  message: string | null
  error_message: string | null
  published_at: string | null
  created_at: string | null
  updated_at: string | null
}

export interface PublicationDetailItem extends PublicationSummaryItem {
  request_payload_json: Record<string, unknown> | unknown[] | null
  response_payload_json: Record<string, unknown> | unknown[] | null
  retry_attempts_total: number
}

export type PublicationListPayload = PaginatedPayload<PublicationSummaryItem>

export interface PublicationDetailPayload {
  publication: PublicationDetailItem | null
  warning: string | null
}

export interface PublicationListFilters {
  article_id?: number
  platform?: string
  status?: string
  trigger_mode?: string
  query?: string
  limit?: number
  offset?: number
}

export type PublicationRetryResult = Record<string, unknown>
